.github/workflows/reports.yml

```
@@ -262,8 +261,145 @@ jobs:
         working-directory: ./test_programs
         run: |
           ./compilation_report.sh
+          cat compilation_report.json
           mv compilation_report.json ../compilation_report.json
 
+      - name: Upload compilation report script
+        uses: actions/upload-artifact@v4
+        with:
+          name: compilation_report_script
+          path: ./test_programs/compilation_report.sh
+          retention-days: 3
+          overwrite: true
+
+      - name: Upload compilation report
+        uses: actions/upload-artifact@v4
+        with:
+          name: in_progress_compilation_report
+          path: compilation_report.json
+          retention-days: 3
+          overwrite: true
+
+  external_repo_compilation_report:
+    needs: [generate_compilation_report]
+    runs-on: ubuntu-latest
+    timeout-minutes: 30
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - project: { repo: AztecProtocol/aztec-packages, path: ./noir-projects/aztec-nr }
+          - project: { repo: AztecProtocol/aztec-packages, path: ./noir-projects/noir-contracts }
+          - project: { repo: AztecProtocol/aztec-packages, path: ./noir-projects/noir-protocol-circuits/crates/parity-lib }
```

```suggestion
          - project: { repo: AztecProtocol/aztec-packages, path: ./noir-projects/noir-contracts }
```

These other two are libraries so we won't get a meaningful compilation time out of them.

---

I have been making some edits locally, so I updated this manually